src/grpc/stream_infer_handler.cc

```diff
@@ -324,12 +324,14 @@ ModelStreamInferHandler::Process(InferHandler::State* state, bool rpc_ok)
     if (err == nullptr) {
       TRITONSERVER_InferenceTrace* triton_trace = nullptr;
 #ifdef TRITON_ENABLE_TRACING
-      GrpcServerCarrier carrier(state->context_->ctx_.get());
-      auto start_options =
-          trace_manager_->GetTraceStartOptions(carrier, request.model_name());
-      state->trace_ = std::move(trace_manager_->SampleTrace(start_options));
-      if (state->trace_ != nullptr) {
-        triton_trace = state->trace_->trace_;
+      if (trace_manager_) {
+        GrpcServerCarrier carrier(state->context_->ctx_.get());
+        auto start_options =
+            trace_manager_->GetTraceStartOptions(carrier, request.model_name());
```

Comment:
I see now. I think this is not the only potential place for a SegFault. Do we want to fix it in other places as well? Or we're targeting `streaming` case at the moment?

Reply:
Would be happy to add checks in other places as well. Could you provide an example where these checks would be needed?

Reply:
@KrishnanPrash Can you look for TRITON_ENABLE_TRACING blocks and break the logic into separate utility function in grpc_utils.h?
This function will take trace_manager_ as input. If trace_manager_ is nullptr then the logic is skipped, otherwise the same logic is run.

Reply:
I think this place (HandleGenerate) seem to be relevant: https://github.com/triton-inference-server/server/blob/dbb064ff930c876b5de46088f675bd60756c2969/src/http_server.cc#L3235-L3237

I can see that HandleInfer on HTTP is guarded: https://github.com/triton-inference-server/server/blob/dbb064ff930c876b5de46088f675bd60756c2969/src/http_server.cc#L3599-L3602

qq to @GuanLuo , I can see that you've added guards for `HandleInfer` (link above), was there a reason not to guard `HandleGenerate` ?

Reply:
@KrishnanPrash, do you need to support trace update with this PR?

Reply:
@oandreeva-nv thank you for pointing it out. Added `nullptr` check for `StartTrace()` in `HandleGenerate()`.

Reply:
@tanmayv25 If it's ok, will move tracing logic to `grpc_utils.cc/h` as helper functions in a follow-up PR [DLIS-7380]

Reply:
> I can see that you've added guards for HandleInfer (link above), was there a reason not to guard HandleGenerate ?

Probably just a missed spot..

Reply:
I see in the HandleInfer() method - that the trace_manager_ check is not guarded with #IF TRACING compile time macro - should it be?

I guess secondary question - if tracing is compiled in - should trace_manager_ be null? - could we make that a pre-condition instead of runtime check if tracing is enabled? (just a question)

Reply:
I don't believe `trace_manager_` needs to be guarded with a `#ifdef TRACING_ENABLE_TRACING` compile time macro because `StartTrace()` function has it's own check for if tracing is enabled. 

As for the secondary question, if tracing is compiled in, `trace_manager_` should technically never be `null` after being passed to the services. However, because the bindings do not yet support tracing, a hopefully temporary situation arises of tracing being enabled, but `trace_manager_` being null.

Reply:
ah so this is temporary? - once enable tracing is added we can remove the runtime check?

Reply:
We could remove the runtime check, but personally I am in favor of keeping these checks because it allows to catch unexpected behavior earlier, rather than later. 

After tracing support is provided in `tritonfrontend`, we could probably modify these checks and return an error to fail earlier with a cleaner error message.